man/sd-journal.xml

```diff
@@ -86,6 +86,16 @@
     — are fully thread-safe and may be called from multiple threads in parallel.</para>
   </refsect1>
 
+  <refsect1>
+    <title>Optional dependencies</title>
```

Comment:
what's the rationale for documenting this? this seems to be somewhat of an implementation detail of things, and we usually don't document these.

Also, isn't this kinda incomplete? Are you sure, sd-jounral is the only API that does this, and that the list is complete?

I'd be more comfortable if this was auto-generated and thus always correct, but this seems a bit random to me?

Reply:
IMO, it'd work better as its own PR. It doesn't fit naturally in this one.

Reply:
I think we *should* add it: if people don't know/think about dlopen, they can be quite confused. It applies to all of libsystemd though. This would fit much better in libsystemd(3), in the NOTES section.

Reply:
The rationale is that with the switch of various libs to dlopen, it will be harder to understand who does what in the public library when external programs link to it and call its APIs. In the abscence of a better way to inspect this, documenting it is the least we can do.

And I put it in sd-journal intentionally to clarify the scope: we know that it's only the sd-journal component of the library that loads these, so it's a safe and stable declaration that we can make, so that users don't get the impression that, say, using sd_bus for their dbus calls also results in these dlopens.